### PR TITLE
Issue 3561/add option to share webform

### PR DESCRIPTION
**Pull request by marvinkome**
#### The solution
Add a share button to share web forms in survey page

#### Screenshots (if appropriate)
image

#### Reviewer Checklist
* [ ] Added an explanation about the work done
* [ ] Connected the PR and the issue on Zenhub
* [ ] Added a test plan to the issue
* [ ] Updated the copyright header (when relevant)
* [ ] Formatted the code
* [ ] Added a documentation (if relevant)
* [ ] Added some unit tests (if relevant)

### CODE REVIEW

File: Dashboard/app/js/lib/views/surveys/form-share.jsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+      this.getProps = this.getProps.bind(this);
+      this.getShareURL = this.getShareURL.bind(this);
+
+      this.valid = false;
+    },
+
+    didInsertElement(...args) {
+      this._super(...args);
+
+      // call validate on init
```
Comment (tangrammer) [inline Dashboard/app/js/lib/views/surveys/form-share.jsx:30]: remove comment and improve format

File: Dashboard/app/js/lib/views/surveys/form-share.jsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+      // case 3 does not have illegal question type
+      const noIllegalQuestion =
+        questions.some(question => {
+          switch (question.get('type')) {
+            case 'CADDISFLY':
+              return true;
+            case 'SIGNATURE':
+              return true;
+            case 'GEOSHAPE':
```
Comment (tangrammer) [inline Dashboard/app/js/lib/views/surveys/form-share.jsx:77]: backend side also checks for CASCADE type https://github.com/akvo/akvo-flow/pull/3564/files#diff-0b87b53604405485c8652761c8c3f11eR28 ... is that correct? @muloem

Comment (muloem) [reply Dashboard/app/js/lib/views/surveys/form-share.jsx:77]: True. That one slipped by me! 🤦  Cascade is supported! 
https://github.com/akvo/akvo-flow/issues/3145#issuecomment-630113567

Comment (marvinkome) [reply Dashboard/app/js/lib/views/surveys/form-share.jsx:77]: Huh, supported or not supported?

Comment (muloem) [reply Dashboard/app/js/lib/views/surveys/form-share.jsx:77]: Webforms can render cascade questions correctly so no need to disable sharing if there is a cascade question.

**Pull request merged**